ReSwift/CoreTypes/Store.swift

```diff
@@ -28,7 +28,7 @@ open class Store<State: StateType>: StoreType {
             subscriptions.forEach {
                 // if a selector is available, subselect the relevant state
                 // otherwise pass the entire state to the subscriber
-                $0.subscriber?._newState(state: $0.selector?(state) ?? state)
+                $0.subscriber?._newState(state: $0.selector?(state) ?? state!)
```

Comment:
@Ben-G I think theres a larger issue at play here introduced by #177 . Was it intentional to allow reducers to return nil? If yes we need to adjust `var state: State!` -> `var state: State?` and make other changes appropriately. If not we should switch reducer back to returning non-nil (which matches the old behaviour) and lets it be safe to force-unwrap the state since it can only be nil before `dispatch(ReSwiftInit())`